File: solutions/lua/armstrong-numbers/2/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
  local digits   = tostring(number)
  local exponent = #digits
  local sum      = 0

  digits:gsub(".", function (digit)
    sum = sum + tonumber(digit)^exponent
  end)

  return number == sum
end

return ArmstrongNumbers

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate)
{
    int exponent = 0;
    int sum = 0;
    int num = candidate;

    do
    {
        exponent++;
        num /= 10;
    } while (num > 0);

    num = candidate;

    for (int i = exponent; i > 0; i--)
    {
        int digit = num % 10;
        num /= 10;
        sum += pow(digit, exponent);
    }

    return candidate == sum;
}
